### PR TITLE
current version

### PR DESCRIPTION
**Pull request by xingdi-eric-yuan**
1. fixed several places related to pytorch 0.4, for example torch.cat vs torch.stack
2. added masks in attentions
3. added more reward functions for policy gradient
4. fixed other bugs.

### CODE REVIEW

**Comment by memray** [comment]
Cool! Thanks.
